### PR TITLE
routing finished, testing in progress

### PR DESCRIPTION
**Pull request by aaronbini**
testing in progress, i'll push again when complete

### CODE REVIEW

File: lib/dummy.json [hunk 1]
```diff
@@ -0,0 +1,272 @@
+[
```
Comment (goodwid) [inline lib/dummy.json]: Not sure what this data is but it looks like real contact data for real people and you should be reticent about submitting such things into a public repo.

File: server.js [hunk 2]
```diff
@@ -1,40 +1,13 @@
...
-handler['/'] = start;
-handler['/goodbye'] = goodbye;
-
-const server = http.createServer(onRequest);
+const server = http.createServer(route.onRequest);
 
 server.listen(8888, () => {
   console.log('Server has started');
-  console.log('Server listening on http://localhost/', PORT);
+  console.log(`Server listening on http://localhost/${PORT}`);
```
Comment (goodwid) [inline server.js]: Incorrect URI.  Should be `http://localhost:${PORT}` with a colon 'twixt the host and port, not a /.

File: package.json [hunk 3]
```diff
@@ -19,7 +19,11 @@
...
   "devDependencies": {
     "chai": "^3.5.0",
-    "mocha": "^2.5.3",
-    "eslint": "^3.1.1"
+    "chai-http": "^3.0.0",
+    "eslint": "^3.1.1",
+    "mocha": "^2.5.3"
+  },
+  "dependencies": {
+    "request": "^2.73.0"
```
Comment (goodwid) [inline package.json]: On line 7, you can change that to `"pretest":"npm run lint",` and add another line `"test":"mocha", which will allow travis to run your tests when you submit the PR.

File: routes.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+  } else {
+    res.write('<p>Hello World.</p><p>Enter "?firstname=[your first name]&lastname=[your last name]" at the end of the url for a tailored hello.</p>');
+  }
+  res.end();
+};
+
+function readFile (req, res) {
+  const pathname = url.parse(req.url).pathname;
+  const querystring = url.parse(req.url, true).query;
+  if (isNotEmpty(querystring) && querystring.format === 'json') {
```
Comment (goodwid) [inline routes.js]: Spiffy!

File: lib/dummy.json (hunk 1 above)
Comment (aaronbini) [inline lib/dummy.json]: this is just dummy data from: http://www.json-generator.com/

**Pull request merged**